l20n/migrate/cldr.py

```diff
@@ -0,0 +1,27 @@
+# coding=utf8
+
+import pkgutil
+import json
+
+
+def in_canonical_order(a, b):
+    return canonical_order.index(a) - canonical_order.index(b)
+
+
+cldr_plurals = json.loads(
+    pkgutil.get_data('l20n.migrate', 'cldr_data/plurals.json')
+)
+
+rules = cldr_plurals['supplemental']['plurals-type-cardinal']
+canonical_order = ('zero', 'one', 'two', 'few', 'many', 'other')
+
+categories = {}
+for lang, rules in rules.items():
+    categories[lang] = sorted(map(
+        lambda key: key.replace('pluralRule-count-', ''),
+        rules.keys()
+    ), in_canonical_order)
+
+
+def get_plural_categories(lang):
+    return categories.get(lang, None)
```

Comment:
I'm not sure what the *reasonable* fallback is.  It might break further down the line because the translation in the `.properties` files might have a different number of plural-specific variants.  Consider `ja-JP-mac`: we don't have an exact match in the `categories` dict for it, so we'd return the fallback: `('one', 'other')`.  However, the translation only has one variant:

    downloadMessage.deleteAll=#1 個のダウンロードを削除しますか？

This variant will become the value of the first category returned by `get_plural_rules`, i.e. `one`.  Which isn't right because `ja` doesn't have a dedicated `one` form in CLDR.

Thanks for starting this topic.  I think the way forward is to make line 27 more strict and raise an exception if `lang` isn't known.  This will make the script exit very early.  For locales like `ja-JP-mac`, the calling party can always call the script with `--lang ja` and maintain the list of similar mapping for other locales, if required.